### PR TITLE
Update for s6 v3 and add apparmor

### PR DESCRIPTION
**Pull request by mdegat01**
Update for base image with s6 v3 (requires new base images before merging). Added an apparmor profile as well. Makes it a better demo and helped to test what changes are required for supervisor apparmor profile.

### CODE REVIEW

**Comment by mdegat01** [comment]
Something of note here. After modifying the apparmor profile to account for the v3 of s6 I also had to `chmod +x` all executables in `services.d`. They were not marked executable before and yet that wasn't a problem with s6 v2. I think that may be something we need to watch out for and note in the blog post.

File: example/apparmor.txt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  capability net_admin,
+  capability net_bind_service,
+  capability dac_read_search,
+  capability dac_override,
+
+  /bin/** ix,
+  /usr/bin/** ix,
+  /usr/lib/bashio/** ix,
+  /etc/s6/** ix,
+  /run/{s6,s6-rc*,service}/** ix,
```
Comment (pvizeli) [inline example/apparmor.txt:20]: Is that not more like:
```suggestion
  /run/s6/** ix,
  /etc/{s6,s6-rc*,service}/** ix,
```

**Review by pvizeli — COMMENTED** [review]
Should we not try to use that structure: https://developers.home-assistant.io/docs/add-ons/presentation/#apparmor

As it match what we document with an RL example + is better readable as this supervise like clone?

File: example/rootfs/etc/services.d/example/finish [hunk 2]
```diff
@@ -6,4 +6,4 @@
 if { s6-test ${1} -ne 0 }
```
Comment (frenck) [inline example/rootfs/etc/services.d/example/finish:6]: We should consider converting this one to Bash too

File: example/apparmor.txt (hunk 1 above)
Comment (mdegat01) [reply example/apparmor.txt:20]: All those folders were in /run when I was poking through the addon.  And that's what kept showing up in the audit when I put the profile in complain mode.  I'm actually not sure if s6 tries to run anything out of /etc anymore, think it copies it everything to /run first

**Comment by mdegat01** [comment]
> Should we not try to use that structure: 

That makes sense. I copied supervisors apparnor profile and started from that since that was the primary concern here. But yea I'll update it to that structure

File: example/rootfs/etc/services.d/example/finish (hunk 2 above)
Comment (mdegat01) [reply example/rootfs/etc/services.d/example/finish:6]: Yea ok that makes sense, will adjust

**Comment by pvizeli** [comment]
The supervisor structure is ugly. Once we should clean that up. I guess that from the documentation is currently the best example and structured

File: example/rootfs/etc/services.d/example/finish (hunk 2 above)
Comment (pvizeli) [reply example/rootfs/etc/services.d/example/finish:6]: then we can also add nice bashio logs :D

File: example/rootfs/etc/services.d/example/run [hunk 3]
```diff
@@ -16,4 +16,4 @@ message=$(bashio::config 'message')
 bashio::log.info "${message:="Hello World..."}"
 
 ## Run your program
-# exec my_program --with-params
+/usr/bin/my_program
```
Comment (pvizeli) [inline example/rootfs/etc/services.d/example/run:19]:
```suggestion
exec /usr/bin/my_program
```
This overwrite the pid/process instead of an child

File: example/rootfs/etc/services.d/example/finish [hunk 4]
```diff
@@ -1,9 +1,9 @@
-#!/usr/bin/execlineb -S1
+#!/usr/bin/bash
```
Comment (pvizeli) [inline example/rootfs/etc/services.d/example/finish:1]:
```suggestion
#!/usr/bin/env bashio
```

File: example/rootfs/etc/services.d/example/finish [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 # Take down the S6 supervision tree when example fails
 # s6-overlay docs: https://github.com/just-containers/s6-overlay
 # ==============================================================================
-if { s6-test ${1} -ne 0 }
-if { s6-test ${1} -ne 256 }
 
-s6-svscanctl -t /var/run/s6/services
+if [[ "$1" -ne 0 ]] && [[ "$1" -ne 256 ]]; then
+  /run/s6/basedir/bin/halt
+fi
```
Comment (pvizeli) [inline example/rootfs/etc/services.d/example/finish:9]:
```suggestion
  bashio::log.warning "Service shutdown"
  /run/s6/basedir/bin/halt
fi
bashio::log.info "Service restart after closing"
```

File: example/rootfs/etc/services.d/example/finish [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
 # ==============================================================================
 # Take down the S6 supervision tree when example fails
 # s6-overlay docs: https://github.com/just-containers/s6-overlay
 # ==============================================================================
-if { s6-test ${1} -ne 0 }
-if { s6-test ${1} -ne 256 }
 
-s6-svscanctl -t /var/run/s6/services
+if [[ "$1" -ne 0 ]] && [[ "$1" -ne 256 ]]; then
+  bashio::log.warning "Service shutdown"
```
Comment (frenck) [inline example/rootfs/etc/services.d/example/finish:8]: Maybe change this to:

```suggestion
  bashio::log.warning "Halt add-on"
```

As its not just the service that stops, the whole container is being taken down.

**Review by frenck — CHANGES_REQUESTED** [review]
This LGTM, I like it 👍

**Review by frenck — APPROVED** [review]
LOL, wrong button, let me try again :D

**Review by pvizeli — APPROVED** [review]
